src/nyx/gpu/gabor.cu

```diff
@@ -0,0 +1,418 @@
+
+        // Wait for device to finish all operation
+        cudaDeviceSynchronize();
+
+        // Check if kernel execution generated and error
+        //getLastCudaError("Kernel execution failed [ solvePoisson ]");
+        cudaError_t err = cudaGetLastError();   
+        if ( err != cudaSuccess ){
+                //fprintf(stderr, "Kernel execution failed [ solvePoisson ]\n");
+                printf("CUDA Error: %s\n", cudaGetErrorString(err));   
+                return;	
+        }
+    }
+
+    __global__ void multiply(CuComplex* A, int row_size, int col_size, CuComplex* B, CuComplex* result, int batch_size) {
+        int i = threadIdx.x + blockIdx.x * blockDim.x;
+        int j = threadIdx.y + blockIdx.y * blockDim.y;
+
+        int index = i*col_size*batch_size+j;
+        double a,b,c,d;
+        if (i < row_size && j < batch_size*col_size) {
+            a = A[index].x;
+            b = A[index].y;
+            c = B[index].x;
+            d = B[index].y;
+
+            result[index].x = a*c - b*d;
+            result[index].y = a*d + b*c;
+        }
+    }
+
+    void cmat_mult(CuComplex* A, int row_size, int col_size, CuComplex* B, CuComplex* result, int batch_size){
+        int block = 16;
+        dim3 threadsPerBlock(block, block);
+        dim3 blocksPerGrid(ceil(row_size/block)+1, ceil(col_size*batch_size/block)+1);
+
+        multiply<<<blocksPerGrid, threadsPerBlock>>>(A, row_size, col_size, B, result, batch_size);
+
+        // Wait for device to finish all operation
+        cudaDeviceSynchronize();
+
+        // Check if kernel execution generated and error
+        //getLastCudaError("Kernel execution failed [ solvePoisson ]");
+        cudaError_t err = cudaGetLastError();   
+        if ( err != cudaSuccess ){
+                //fprintf(stderr, "Kernel execution failed [ solvePoisson ]\n");
+                printf("CUDA Error: %s\n", cudaGetErrorString(err));   
+                return;	
+        }
+    }
+
+    void conv_dud_gpu_fft(double* out, 
+                            const unsigned int* image, 
+                            double* kernel, 
+                            int image_n, int image_m, int kernel_n, int kernel_m){
+
+        
+        
+        int batch_size = 1;
+
+        // calculate new size of image based on padding size
+        int row_size = image_m + kernel_m - 1;
+        int col_size = image_n + kernel_n - 1;
+        int size = row_size * col_size;
+
+        if ((2 * size * batch_size) >= CUFFT_MAX_SIZE) {
+            throw invalid_argument("Batch of images is too large. The maximumum number of values in cuFFT is 2^27.");
+        }
+
+        // allocate space for linear indexed arrays
+        Complex* linear_image = (Complex*)malloc(size * batch_size * sizeof(Complex));
+        Complex* result = (Complex*)malloc(size * batch_size * sizeof(Complex));
+        Complex* linear_kernel = (Complex*)malloc(size * batch_size * sizeof(Complex));
```

Comment:
We should add check for allocation failure. Also, can STL containers (vector?)  be used here?